modules/taboolaIdSystem.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * This module adds Taboola’s User ID submodule to the Prebid User ID module.
+ * @module modules/taboolaIdSystem
+ * @requires module:modules/userId
+ */
+
+import {submodule} from '../src/hook.js';
+import {ajax} from '../src/ajax.js';
+import {getStorageManager} from '../src/storageManager.js';
+import {logError} from '../src/utils.js';
+import {gdprDataHandler, uspDataHandler, gppDataHandler} from '../src/adapterManager.js';
+import {MODULE_TYPE_UID} from '../src/activities/modules.js';
+
+/**
+ * The Taboola sync endpoint.
+ * NOTE: We already have query params (?app.type=desktop&app.apikey=...),
+ * so we'll handle extra params carefully.
+ */
+const TABOOLA_SYNC_ENDPOINT = 'https://api.taboola.com/1.2/json/taboola-usersync/user.sync?app.type=desktop&app.apikey=e60e3b54fc66bae12e060a4a66536126f26e6cf8';
+const BIDDER_CODE = 'taboola';
+const GVLID = 42;
+const STORAGE_KEY = 'taboola global:user-id';
+// Taboola cookie keys
+const COOKIE_KEY = 'trc_cookie_storage';
+const TGID_COOKIE_KEY = 't_gid';
+const TGID_PT_COOKIE_KEY = 't_pt_gid';
+const TBLA_ID_COOKIE_KEY = 'tbla_id';
+export const sm = getStorageManager({
+  moduleType: MODULE_TYPE_UID,
+  moduleName: BIDDER_CODE
+});
+
+/**
+ *  Taboola’s ID retrieval logic.
+ * - Local storage
+ * - Known Taboola cookies
+ * - window.TRC.user_id
+ */
+const userData = {
+  getUserId() {
+    try {
+      return this.getFromLocalStorage() || this.getFromCookie() || this.getFromTRC();
+    } catch (ex) {
+      return '0';
+    }
+  },
+
+  getFromLocalStorage() {
+    const {hasLocalStorage, localStorageIsEnabled, getDataFromLocalStorage} = sm;
+    if (hasLocalStorage() && localStorageIsEnabled()) {
+      return getDataFromLocalStorage(STORAGE_KEY);
+    }
+    return undefined;
+  },
+
+  getFromCookie() {
+    const {cookiesAreEnabled, getCookie} = sm;
+    if (cookiesAreEnabled()) {
+      const mainCookieData = getCookie(COOKIE_KEY);
+      if (mainCookieData) {
+        const userId = this.getCookieDataByKey(mainCookieData, 'user-id');
+        if (userId) {
+          return userId;
+        }
+      }
+      // Fallback checks
+      const tid = getCookie(TGID_COOKIE_KEY);
+      if (tid) {
+        return tid;
+      }
+      const tptId = getCookie(TGID_PT_COOKIE_KEY);
+      if (tptId) {
+        return tptId;
+      }
+      const tblaId = getCookie(TBLA_ID_COOKIE_KEY);
+      if (tblaId) {
+        return tblaId;
+      }
+    }
+    return undefined;
+  },
+
+  /**
+   * Extract a value for a given key out of a multi-value cookie, e.g. "user-id=abc&foo=bar".
+   */
+  getCookieDataByKey(cookieData, key) {
+    if (!cookieData) {
+      return undefined;
+    }
+    const [match] = cookieData.split('&').filter(item => item.startsWith(`${key}=`));
+    if (match) {
+      // eslint-disable-next-line no-unused-vars
+      const [_, value] = match.split('=');
```

Comment:
```suggestion
      const value = match.split('=')[1];
```
remove 
// eslint-disable-next-line no-unused-vars